timely/src/progress/operate.rs

```diff
@@ -58,6 +58,55 @@ pub trait Operate<T: Timestamp> : Schedule {
     fn notify_me(&self) -> bool { true }
 }
 
+/// Operator internal connectivity, from inputs to outputs.
+pub type Connectivity<TS> = Vec<PortConnectivity<TS>>;
+/// Internal connectivity from one port to any number of opposing ports.
+#[derive(serde::Serialize, serde::Deserialize, columnar::Columnar, Debug, Clone, Eq, PartialEq)]
+pub struct PortConnectivity<TS> {
+    tree: std::collections::BTreeMap<usize, Antichain<TS>>,
+}
+
+impl<TS> Default for PortConnectivity<TS> {
+    fn default() -> Self {
+        Self { tree: std::collections::BTreeMap::new() }
+    }
+}
+
+impl<TS> PortConnectivity<TS> {
+    /// Inserts an element by reference, ensuring that the index exists.
+    pub fn insert(&mut self, index: usize, element: TS) -> bool where TS : crate::PartialOrder + Clone {
```

Comment:
```suggestion
    pub fn insert(&mut self, index: usize, element: TS) -> bool where TS : crate::PartialOrder {
```
Nit: We don't need `Clone` here.